src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringInterpolation.cs

```diff
@@ -33,6 +32,42 @@ private BoundExpression RewriteInterpolatedStringConversion(BoundConversion conv
             return result;
         }
 
+        private static bool CanLowerToStringConcatenation(BoundInterpolatedString node)
+        {
+            foreach (var part in node.Parts)
+            {
+                if (part is BoundStringInsert fillin)
+                {
+                    // this is one of the expression holes
+                    if (fillin.HasErrors ||
+                        fillin.Value.Type?.SpecialType != SpecialType.System_String ||
```

Comment:
can't we box other types e.g. int and pass to `Concat(object, object)` etc?

Reply:
@alrz I think the purpose of this PR is to get the camel's nose into the tent.

Reply:
![image](https://user-images.githubusercontent.com/3804346/40033858-1a6e3812-57af-11e8-9a99-e5f066145153.png)

Reply:
I suppose that's a no -- due to custom formatting semantics. camel's nose shall not get into the tent.

Reply:
What I meant was doing this for strings *is* the camel's nose in the tent. Once this optimization is done, if we can demonstrate that it would be correct to do the optimization for other types in the future, the optimization could be extended in the future.